jobAPI_swagger.yaml

```diff
@@ -0,0 +1,378 @@
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Job'          
+        '400':
+          description: Invalid ID supplied
+        '404':
+          description: Job not found
+    delete:
+      tags:
+        - admin
+      summary: Delete a particular job
+      description: Only deletes on the platform, not communication channels
+      operationId: deleteJob
+      parameters:
+        - name: jobId
+          in: path
+          description: ID of job to delete
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '204':
+          description: Job deleted successfully
+        '404':
+          description: Job not found
+    put:
+      tags:
+        - admin
+      summary: Update an existing job
+      description: Update an existing job by ID
+      operationId: updateJob
+      parameters:
+        - name: jobId
+          in: path
+          description: ID of job to update
+          required: true
+          schema:
+            type: integer
+            format: int64
+      requestBody:
+        description: Update an existing job in the system
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Job'
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Job'          
+        '400':
+          description: Invalid ID supplied
+        '404':
+          description: Job not found
+        '422':
+          description: Validation exception
+  /jobs:
+    get:
+      tags:
+        - job
+      summary: Get jobs with filters
+      description: Returns a list of jobs based on filters
+      operationId: getJobs
+      parameters:
+        - name: categoryId
+          in: query
+          description: Filter by category ID
+          schema:
+            type: integer
+            format: int64
+        - name: levelId
+          in: query
+          description: Filter by level ID
+          schema:
+            type: integer
+            format: int64
+        - name: companyName
+          in: query
+          description: Filter by company name
+          schema:
+            type: string
+        - name: popular
+          in: query
+          description: Filter by popular jobs
+          schema:
+            type: boolean
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Job'
+        '404':
+          description: No jobs found for the given filters
+  /job/submit:
```

Comment:
drop the /submit
POST /jobs already refers that a job is submitted for creation